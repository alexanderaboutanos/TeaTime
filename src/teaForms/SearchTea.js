/** @format */
import { useState } from "react";
import { Button } from "react-bootstrap";

const SearchTea = ({ search }) => {
  const INITIAL_STATE = { search: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  /** Send {name, quantity} to parent
   *    & clear form. */

  const handleSubmit = (evt) => {
    evt.preventDefault();
    search(formData);
    setFormData(INITIAL_STATE);
  };

  /** Update local state w/curr state of input elem */

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  /** render form */

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label htmlFor="search"></label>
        <input
          className="form-control"
          style={{ marginBottom: "15px" }}
          placeholder="type your search here..."
          id="search"
          name="search"
          value={formData.search}
          onChange={handleChange}
        />
        <Button style={{ marginBottom: "15px" }}>Search</Button>
      </form>
    </div>
  );
};

export default SearchTea;
