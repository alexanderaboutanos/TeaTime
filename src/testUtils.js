/** @format */

import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser1",
  first_name: "Test",
  last_name: "User",
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

const fakeTeas = [
  {
    id: 740259,
    title:
      "Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea",
    brand: "Chinese Tea Culture",
    description:
      "Tea Sampler - Fruit Tea - Antioxidants Tea - Rooibos Tea - 100% Natural - Decaffeinated - Blueberry Tea - Peach Tea - Raspberry Tea - Loose Leaf Tea",
    img_url: "https://spoonacular.com/productImages/740259-312x231.jpeg",
  },
  {
    id: 1539729,
    title:
      "Tea Forte Lotus Relaxing Teas Presentation Box Tea Sampler Gift Set, 20 Assorted Variety Handcrafted Pyramid Tea Infuser Bags, Black Tea, Green Tea, Oolong Tea, White Tea, Herbal Tea",
    brand: "Tea Forte",
    description:
      "LOTUS COLLECTION, a curated assortment of five relaxing hot teas for mind, body & spirit - a unique and unprecedented gourmet collection of highest-grade, small-lot, certified organic teas: Vanilla Pear, Orange Jasmine, Mountain Oolong, Darjeeling Quince, Lemon Lavender DIVERSE, SUBTLE FLAVORS from only the finest whole tea leaves in the world, blended with aromatic fruits, flowers and herbs DELIGHTFUL TEA GIFT SET offers a variety of premium gourmet teas, a most welcomed hostess gift or gift for tea lovers PRESENTATION BOX tea sampler has 20 assorted pyramid tea bag infusers, handcrafted to allow the delicate leaves to luxuriantly unfurl in hot water, producing a deliciously aromatic, flavorful cup WRAPPED IN ELEGANT, embossed pattern gift box with a satin ribbon, open the lid to reveal a tea tasting menu with blend descriptions",
    img_url: "https://spoonacular.com/productImages/1539729-312x231.jpeg",
  },
];

export { UserProvider, fakeTeas };
