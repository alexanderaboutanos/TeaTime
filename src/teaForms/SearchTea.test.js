/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import SearchTea from "./SearchTea";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <SearchTea />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <SearchTea />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
