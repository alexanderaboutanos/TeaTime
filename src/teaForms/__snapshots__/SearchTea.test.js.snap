// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="search"
      />
      <input
        class="form-control"
        id="search"
        name="search"
        placeholder="type your search here..."
        style="margin-bottom: 15px;"
        value=""
      />
      <button
        class="btn btn-primary"
        style="margin-bottom: 15px;"
        type="button"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
